definitions/checks/foreman_proxy/check_tftp_storage.rb

```diff
@@ -42,12 +42,8 @@ def token_duration
     end
 
     def lookup_token_duration
-      data = feature(:foreman_database). \
-             query("select s.value, s.default from settings s \
-                    where category IN ('Setting::Provisioning','Setting') \
-                    and name = 'token_duration'")
-
-      YAML.load(data[0]['value'] || data[0]['default'])
+      data = feature(:hammer).run('--output yaml settings info --id token_duration')
+      YAML.load(data['Value'])
```

Comment:
Shouldn't this be (untested)
```suggestion
      YAML.load(data)['Value']
```
As `data` will be the YAML hammer returns, which you need to parse and then get the Value out of it?

Reply:
And if you wonder why this seems to work:

```
irb(main):001:0> a = "dgdfg Value sgdfg"
=> "dgdfg Value sgdfg"
irb(main):002:0> a['Value']
=> "Value"
```

So you get `Value` as the return of `lookup_token_duration`

Reply:
Nutty

Reply:
Ruby, but I'll take yours too.